File: 217-contains-duplicate/217-contains-duplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        # Initialize a hash-set
        exists = set()
        
        for num in nums:
            # If the number already exists in the hash-set return True.
            if num in exists: return True
            # Otherwise add it to the hash set.
            exists.add(num)
        
        # If no duplicates exists return False.
        return False

File: 217-contains-duplicate/217-contains-duplicate.cpp
#include <unordered_set>

using namespace std;

class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_set<int> exists;
        for (auto num : nums) {
            if (exists.count(num))
                return true;
            exists.insert(num);
        }
        return false;
    }
};